concepts/pointers/about.md

```diff
@@ -1,3 +1,154 @@
+```
+
+### Accessing the value via a pointer (dereferencing)
+
+When we have a pointer, we might want to know the value stored in the memory address to which it points. We can do this using the `*` operator:
+
+```go
+var a int 
+a = 2
+
+var p *int 
+p = &a // the variable 'p' will hold the memory address of 'a'
+
+var b int
+b = *p // b == 2
+```
+
+The operation `*p` fetches the value stored at the memory address stored in `p`. This operation is often called "dereferencing".
+
+We can also use the derefering operator to assign a new value to the memory address referenced by the pointer:
+
+```go
+var a int
+a = 2   // declare int variable 'a' and assign it the value of 2
+
+var pa *int
+pa = &a          // 'pa' now refers to the memory address of 'a'  
+*pa = *pa + 2    // increment by 2 the value at memory address 'pa'
+
+fmt.Println(a)   // Output: 4
+                 // 'a' will have the new value that was changed via the pointer!
+```
+
+Assigning to `*pa` will change the value stored at the memory address `pa` holds. Since `pa` holds the memory address of `a`, by assigning to `*pa` we are effectively changing the value of `a`! 
+
+A note of caution however: always check if a pointer is not `nil` before dereferencing. Dereferencing a `nil` pointer will make the program crash at runtime!
+
+### Pointers to structs
+
+So far we've only seen pointers to primitive values. We can also create pointers for structs:
+
+```go
+type Person struct {
+    Name string
+    Age  int
+}
+
+var peter Person
+peter = Person{Name: "Peter", Age: 22}
+
+var p *Person
+p = &peter
+```
+
+We could have also created a new `Person` and immediately stored a pointer to it:
+
+```go
+var p *Person
+p = &Person{Name: "Peter", Age: 22}
+```
+
+When we have a pointer to a struct, we don't need to dereference the pointer before accessing one of the fields:
+
+```go
+var p *Person
+p = &Person{Name: "Peter", Age: 22}
+
+fmt.Println(p.Name) // Output: "Peter" 
+                    // Go automatically dereferences 'person' to allow
+                    // access to the 'Name' field
+```
+
+## Slices and maps are already pointers
+
+Slices and maps are special types because they already have pointers in their implementation. This means that more often that not, we don't need to create pointers for these types to share the memory address for their values. Imagine we have a function that adds a key to a map:
+
+
+```go
+func addPeterAge(m map[string]int){
+	m["Peter"] = 22
+}
+```
+
+If we create a map and call this function, the changes the function made to the map persist after the function ended. This is a similar behavior we get if we were using a pointer, but note how on this example we are not using any referencing/dereferencing or any of the pointer syntax:
+
+```go
+ages := make(map[string]int)
+addPeterAge(ages)
+fmt.Println(ages)
+// Output: map[Peter:22]
+// The changes the function 'addPeterAge' made to the map are visible after the function ends!
+```
+
```

Comment:
Suggestion how to address the slices issue mentioned here https://github.com/exercism/go/pull/1724#issuecomment-932351399
```suggestion

The same applies when changing an existing item in a slice. However, actions that return a new slice like `append` will **not** modify the slice outside of the function.
```

Reply:
This is actually a common edge case in go.
```
func addOne(in []int) []int {
	return append(in, 1)
}

func main() {
	slice := []int{1, 2, 3, 4}
	slice = append(slice, 5)
	slice2 := append(slice, 6)
	fmt.Println(addOne(slice)) // prints 1,2,3,4,5,1
	fmt.Println(slice2)        // also prints 1,2,3,4,5,1
}
```